cpp/src/ast/expressions.cpp

```diff
@@ -26,33 +26,35 @@
 namespace cudf {
 namespace ast {
 
-operation::operation(ast_operator op, expression const& input) : op(op), operands({input})
+operation::operation(ast_operator op, expression const& input)
+  : op{op}, operands{input, input}, arity{1}
 {
-  if (cudf::ast::detail::ast_operator_arity(op) != 1) {
-    CUDF_FAIL("The provided operator is not a unary operator.");
-  }
+  CUDF_EXPECTS(cudf::ast::detail::ast_operator_arity(op) == 1,
+               "The provided operator is not a unary operator.");
```

Comment:
*suggestion*: Shall we throw `std::invalid_argument` here?

Reply:
done. thanks!